docs/src/CommutativeAlgebra/rings.md

```diff
@@ -42,36 +45,83 @@ order). The other possible choices are `:deglex` and `:degrevlex`. Gröbner base
 
 ###### Examples
 
-```@repl oscar
-R, (x, y, z) = PolynomialRing(ZZ, ["x", "y", "z"])
-typeof(R)
-typeof(x)
-S, (x, y, z) = PolynomialRing(ZZ, ["x", "y", "z"])
-R === S
+```jldoctest
+julia> R, (x, y, z) = PolynomialRing(ZZ, ["x", "y", "z"])
+(Multivariate Polynomial Ring in x, y, z over Integer Ring, fmpz_mpoly[x, y, z])
+
+julia> typeof(R)
+FmpzMPolyRing
+
+julia> typeof(x)
+fmpz_mpoly
+
+julia> S, (x, y, z) = PolynomialRing(ZZ, ["x", "y", "z"])
+(Multivariate Polynomial Ring in x, y, z over Integer Ring, fmpz_mpoly[x, y, z])
+
+julia> R === S
+true
+
 ```
 
-```@repl oscar
-R1, x = PolynomialRing(QQ, ["x"])
-typeof(x)
-R2, (x,) = PolynomialRing(QQ, ["x"])
-typeof(x)
-R3, x = PolynomialRing(QQ, "x")
-typeof(x)
+```jldoctest
+julia> R1, x = PolynomialRing(QQ, ["x"])
+(Multivariate Polynomial Ring in x over Rational Field, fmpq_mpoly[x])
+
+julia> typeof(x)
+fmpq_poly
+fmpq_mpoly
+Vector{fmpq_mpoly} (alias for Array{fmpq_mpoly, 1})
+
+julia> R2, (x,) = PolynomialRing(QQ, ["x"])
+(Multivariate Polynomial Ring in x over Rational Field, fmpq_mpoly[x])
+
+julia> typeof(x)
+
+julia> R3, x = PolynomialRing(QQ, "x")
+(Univariate Polynomial Ring in x over Rational Field, x)
+
+julia> typeof(x)
+
 ```
 
-```@repl oscar
-V = ["x[1]", "x[2]"]
-T, x = PolynomialRing(GF(3), V)
-x
+```julia
+julia> V = ["x[1]", "x[2]"]
+2-element Vector{String}:
+ "x[1]"
+ "x[2]"
+
+julia> T, x = PolynomialRing(GF(3), V)
+2-element Vector{gfp_mpoly}:
+ x[1]
+ x[2] = PolynomialRing(GF(3), ["x[1]", "x[2]"])
+(Multivariate Polynomial Ring in x[1], x[2] over Galois field with characteristic 3, gfp_mpoly[x[1], x[2]])
+
+julia> x
+
 ```
 
 The constructor illustrated below allows for the convenient handling of variables with multi-indices:
 
-```@repl oscar
-R, x, y, z = PolynomialRing(QQ, "x" => (1:3, 1:4), "y" => 1:2, "z" => (1:1, 1:1, 1:1))
-x
-y
-z
+```julia
+julia> R, x, y, z = PolynomialRing(QQ, "x" => (1:3, 1:4), "y" => 1:2, "z" => (1:1, 1:1, 1:1))
+3×4 Matrix{fmpq_mpoly
+2-element Vector{fmpq_mpoly}:
+ y[1]
+ y[2]}:
+ x[1, 1]  x[1, 2]  x[1, 3]  x[1, 4]
+ x[2, 1]  x[2, 2]  x[2, 3]  x[2, 4]
+ x[3, 1]  x[3, 2]  x[3, 3]  x[3, 4], y, z
+1×1×1 Array{fmpq_mpoly, 3}:
+[:, :, 1] =
+ z[1, 1, 1] = PolynomialRing(QQ, "x" => (1:3, 1:4), "y" => 1:2, "z" => (1:1, 1:1, 1:1))
+(Multivariate Polynomial Ring in 15 variables x[1, 1], x[2, 1], x[3, 1], x[1, 2], ..., z[1, 1, 1] over Rational Field, fmpq_mpoly[x[1, 1] x[1, 2] x[1, 3] x[1, 4]; x[2, 1] x[2, 2] x[2, 3] x[2, 4]; x[3, 1] x[3, 2] x[3, 3] x[3, 4]], fmpq_mpoly[y[1], y[2]], fmpq_mpoly[z[1, 1, 1]])
+
+julia> x
+
+julia> y
+
+julia> z
+
```

Comment:
same as above: The output should be rearranged, and the block could be a `jldoctest`.